src/rng.rs

```diff
@@ -485,6 +516,25 @@ mod test {
         assert_eq!(array, gen);
     }
 
+    #[test]
+    fn test_gen_range_uninit_same_as_simple() {
+        let mut r = rng(101);
+        let mut mu_array: [mem::MaybeUninit<char>; 64] = unsafe {
+            // Safe to assume init because inner values have uninit type.
+            mem::MaybeUninit::uninit().assume_init()
```

Comment:
Did you read the doc on [`assume_init`](https://doc.rust-lang.org/stable/std/mem/union.MaybeUninit.html#method.assume_init)? This is not safe.

It sounds like you want to cast `[MaybeUninit<T>; N]` to `&mut [T]`. This is explicitly not allowed by the Rust memory model.

Reply:
Nope, you are wrong.

This code is same as in MaybeUninit docs actually.
https://doc.rust-lang.org/stable/std/mem/union.MaybeUninit.html#initializing-an-array-element-by-element

And I make an array of `MaybeUninit`s which can be created from uninit array.

Reply:
Correct. It's just your comment that is wrong then: the reason it is safe is that the *outer* type is an array of `MaybeUninit`.